### PR TITLE
docs(examples): Add "multipart/form-data" example

### PR DESCRIPTION
**Pull request by David200308**
#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by mcollina — APPROVED** [review]
lgtm

File: examples/parser.js [hunk 1]
```diff
@@ -3,6 +3,11 @@
 const fastify = require('../fastify')({ logger: true })
 const jsonParser = require('fast-json-body')
 const querystring = require('node:querystring')
+const fs = require('fs');
+const os = require('os');
+const path = require('path');
```
Comment (Fdawgs) [inline examples/parser.js:8]:
```suggestion
const fs = require('node:fs');
const os = require('node:os');
const path = require('node:path');
```

**Review by Fdawgs — CHANGES_REQUESTED** [review]

**Review by Uzlopak — CHANGES_REQUESTED** [review]
We should dog food our own @fastify/busboy

File: examples/parser.js [hunk 2]
```diff
@@ -18,7 +22,7 @@ fastify.addContentTypeParser('application/jsoff', function (request, payload, do
 fastify.addContentTypeParser('application/x-www-form-urlencoded', function (request, payload, done) {
   let body = ''
   payload.on('data', function (data) {
-    body += data
+    body += data.toString()
```
Comment (Uzlopak) [inline examples/parser.js:25]: hmm. I guess urlencoded si pure ascii, but if it is multibyte encoded, than this bodyparser could be error prone.

Should be imho:

```js
fastify.addContentTypeParser('application/x-www-form-urlencoded', function (request, payload, done) {
  const body = []
  payload.on('data', function (data) {
    body.push(data)
  })
  payload.on('end', function () {
    try {
      const parsed = querystring.parse(Buffer.concat(body).toString())
      done(null, parsed)
    } catch (e) {
      done(e)
    }
  })
  payload.on('error', done)
})

```

**Review by climba03003 — CHANGES_REQUESTED** [review]
I would against providing a simple `multipart/form-data` example for everyone.
People always copy and paste the example and it is a critical one that is not safe for copy and paste.

**Pull request closed without merging**